<HTML>
<HEAD>
<meta charset="UTF-8">
<title>NestedExceptionUtils.getMostSpecificCause - spring-framework</title>
<link rel="stylesheet" href="../../../style.css">
</HEAD>
<BODY>
<a href="../../index.html">spring-framework</a>&nbsp;/&nbsp;<a href="../index.html">org.springframework.core</a>&nbsp;/&nbsp;<a href="index.html">NestedExceptionUtils</a>&nbsp;/&nbsp;<a href=".">getMostSpecificCause</a><br/>
<br/>
<h1>getMostSpecificCause</h1>
<a name="org.springframework.core.NestedExceptionUtils$getMostSpecificCause(kotlin.Throwable)"></a>
<code><span class="keyword">open</span> <span class="keyword">static</span> <span class="keyword">fun </span><span class="identifier">getMostSpecificCause</span><span class="symbol">(</span><span class="identifier" id="org.springframework.core.NestedExceptionUtils$getMostSpecificCause(kotlin.Throwable)/original">original</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a></code>
<p>Retrieve the most specific cause of the given exception, that is, either the innermost cause (root cause) or the exception itself. 
<p>Differs from <code><a href="#">#getRootCause</a></code> in that it falls back to the original exception if there is no root cause.</p>
</p>
<h3>Parameters</h3>
<p><a name="original"></a>
<code>original</code> - the original exception to introspect</p>
<p><strong>Return</strong><br/>
the most specific cause (never <code>null</code>)</p>
<p><strong>Since</strong><br/>
4.3.9</p>
</BODY>
</HTML>
